File: old_groovy/src/main/groovy/net/zomis/brainf/model/ast/BFToken.java
package net.zomis.brainf.model.ast;

import net.zomis.brainf.model.BrainfuckCommand;
import net.zomis.brainf.model.classic.BrainFCommand;

public class BFToken extends Token {

    public final BrainfuckCommand command;

    public BFToken(TokenInfo tokenInfo, BrainFCommand bfCommand) {
        super(tokenInfo);
        this.command = bfCommand;
    }

    @Override
    public String toString() {
        return "BFToken{" +
                "command=" + command +
                "} " + super.toString();
    }

}

File: brainduck-kotlin/src/commonMain/kotlin/net/zomis/brainf/model/ast/BFToken.kt
package net.zomis.brainf.model.ast;

import net.zomis.brainf.model.classic.BrainFCommand;

class BFToken(
    tokenInfo: TokenInfo,
    val command: BrainFCommand,
) : Token(tokenInfo) {

    override fun toString(): String {
        return "BFToken{" +
                "command=" + command +
                "} " + super.toString();
    }

}
